src/dmd/dtemplate.d

```diff
@@ -6365,12 +6365,15 @@ extern (C++) class TemplateInstance : ScopeDsymbol
                         .error(loc, "`%s` is not a valid template instance, because `%s` is not a template declaration but a type (`%s == %s`)", toChars(), id.toChars(), id.toChars(), s.getType.kind());
                         return false;
                     }
-                    s = s2;
+                    // because it can be the alias created for a TemplateParameter
+                    if (!s.isAliasDeclaration() /*&& !s.isAliasCreatedForOneTemplateParameter*/)
```

Comment:
Ideally I would add a new class (TemplateAliasDeclaration, derived from AliasDeclaration) because testing for alias catches the case but may introduce regressions of type reject-valid this time.

Reply:
at least a new field ...